### PR TITLE
Always show full name and version when running a valid dnu subcommand

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/1864

@lodejard @muratg @davidfowl

### CODE REVIEW

**Comment by ChengTian** [comment]
Sample output:
showfulltitle

**Comment by ChengTian** [comment]
Show the header info only when a subcommand is actually running. If you provide an invalid subcommand/argument/option, we don't show the header:

showfulltitle_error

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs [hunk 1]
```diff
@@ -432,6 +428,23 @@ public void ShowVersion()
             Console.WriteLine(VersionGetter());
         }
 
+        public string GetFullNameAndVersion()
+        {
+            return VersionGetter == null ? FullName : string.Format("{0} v{1}", FullName, VersionGetter());
```
Comment (ChengTian) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: This file will be compiled by Mono msc compiler, which doesn't support string interpolation. So using `string.Format()` here

**Comment by BrennanConroy** [comment]
:shipit:

**Pull request merged**